packages/context-editor/src/components/AttributePanel.tsx

```diff
@@ -1,221 +1,253 @@
 import type { Mark } from "prosemirror-model";
 
 import React, { useEffect, useState } from "react";
-import { EditorView } from "prosemirror-view";
+import {
+	useEditorEffect,
+	useEditorEventCallback,
+	useEditorState,
+} from "@handlewithcare/react-prosemirror";
 
 import { Input } from "ui/input";
 import { Label } from "ui/label";
 
-import type { PanelProps } from "../ContextEditor";
+import { useEditorContext } from "./Context";
+import { MENU_BAR_HEIGHT } from "./MenuBar";
 
 const animationTimeMS = 150;
 const animationHeightMS = 100;
 
-export interface AttributePanelProps {
-	panelPosition: PanelProps;
-	viewRef: React.RefObject<EditorView | null>;
+interface PanelProps {
+	top: number;
+	left: number;
+	right: number | string;
+	bottom: number;
 }
 
-export function AttributePanel({ panelPosition, viewRef }: AttributePanelProps) {
-	const [position, setPosition] = useState(panelPosition);
-	/* Set as init position and then keep track of state here, while syncing 
-	so the panel doesn't become out of sync with doc (only an issue if values are shown
-	and edited elsewhere */
+const initPanelProps: PanelProps = {
+	top: 0,
+	left: 0,
+	right: "100%",
+	bottom: 0,
+};
+
+export function AttributePanel({ menuHidden }: { menuHidden: boolean }) {
+	const [position, setPosition] = useState(initPanelProps);
 	const [height, setHeight] = useState(0);
+	const state = useEditorState();
+	const {
+		activeNode,
+		position: activeNodePosition,
+		setActiveNode,
+		setPosition: setActiveNodePosition,
+	} = useEditorContext();
+
+	/**
+	 * This determination of the 'activeNode' is prone to bugs. We should figure
+	 * out a better way to do it.
+	 **/
```

Comment:
right now this mostly works, but there are edge cases such as if the user is typing **bold** and then they unbold, the attribute panel might stay open

https://github.com/user-attachments/assets/9b09be1a-e19b-41e1-b790-9b3a46b54aba

the most important thing is that if a paragraph has content, and we try to edit the paragraph's attributes, the panel closes automatically. this is because this bit of code thinks we are in the next node, which is no good! I will try to fix this soon

here it is working in other cases though

https://github.com/user-attachments/assets/5ca9c15e-3ffe-425d-891c-23bf6d672b4e

Reply:
there does seem to be a big with the updating mark logic that also seems to exist with the editor prior to this. I haven't been able to figure it out yet, but it seems like the mark attrs will sometimes apply to _all_ mark attributes. this does not appear to be the case if you already have a document loaded and are applying marks there though (i.e. it works ok in the ContextEditor > Primary story)

https://github.com/user-attachments/assets/4aa367db-cd74-40ac-b884-c77a6afb38a5